nss/aad_auth/testdata/golden/TestGetEnt/try_to_list_group_without_any_cache

```diff
@@ -0,0 +1,2 @@
+|
```

Comment:
Good that we have tests :)

without any cache this should return the numeric equivalent of:

status: NSS_STATUS_UNAVAIL
errno: ENOENT

(https://www.gnu.org/software/libc/manual/html_node/NSS-Modules-Interface.html).
(Note: I’more more sure in this case that we return the right thing in `internal/nss/` layer, but worth checking. You can see this is the return value when we try to get something from shadow without having access to it).

Reply:
Again, the problem is likely on the internal layer. When we call the function to [StartEntryIteration](https://github.com/ubuntu/aad-auth/blob/f98df3993b8619c5cad4bdfb4a0b5d38e568d3c8/internal/nss/group/group.go#L88), it calls cache.New(). The problem is: cache.new() also calls [initDB](https://github.com/ubuntu/aad-auth/blob/f98df3993b8619c5cad4bdfb4a0b5d38e568d3c8/internal/cache/cache.go#L204) and initDB creates the databases needed if they don't exist.
Summarizing, there is no way for us to have a "no cache" case, only "empty cache".

Reply:
We can have a test this way: if we can’t create the db.

So, the test is: "no database file and no permission to create it", mind having a shot?